### PR TITLE
Architecture: CallFrame/Substate Refactor

### PR DESCRIPTION
**Pull request by talekhinezh**
This PR continues the Substate refactoring:
* Remove `RENodeInfo` and replaced `visibility` flag with an actor to substate_id visibility mapping more closely resembling crate visibilities
* Reduced granularity of borrowing data to substates
* Implement simple global component reference passing, removed ability to access `ComponentInfo` of any global component at any time.

### CODE REVIEW

**Review by iamyulong — COMMENTED** [review]
I might've missed it, but where do we add component refs for newly globalised component?

File: radix-engine/src/engine/call_frame.rs [hunk 1]
```diff
@@ -1342,8 +1381,56 @@ where
...
+
+            // Collect component addresses
+            for component_address in &input.refed_component_addresses {
+                component_addresses.insert(*component_address);
+            }
+            let input: TransactionProcessorRunInput = scrypto_decode(&input.raw).unwrap();
+            for instruction in &input.instructions {
+                match instruction {
+                    ExecutableInstruction::CallFunction { arg, .. }
+                    | ExecutableInstruction::CallMethod { arg, .. } => {
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply radix-engine/src/engine/call_frame.rs:1408]: Good catch

File: radix-engine/src/engine/call_frame.rs [hunk 2]
```diff
@@ -2374,12 +2492,16 @@ where
             }
         }
 
+        self.node_refs
+            .insert(node_id, RENodePointer::Store(node_id));
```
Comment (talekhinezh) [inline radix-engine/src/engine/call_frame.rs:2496]: @iamyulong Globalized nodes refed here

File: radix-engine/src/engine/call_frame.rs [hunk 3]
```diff
@@ -1342,8 +1381,56 @@ where
...
+                    *package_address,
+                    blueprint_name.clone(),
+                ))
+            }
+            TypeName::Package | TypeName::ResourceManager | TypeName::TransactionProcessor => {
+                REActor::Native
+            }
+        };
+
+        // Move this into transaction processor
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply radix-engine/src/engine/call_frame.rs:1395]: Sure makes sense. Either way, should be another layer.

File: radix-engine/src/engine/call_frame.rs (hunk 2 above)
Comment (iamyulong) [reply radix-engine/src/engine/call_frame.rs:2496]: Nice, thanks!

**Review by iamyulong — APPROVED** [review]

File: radix-engine/src/engine/call_frame.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (talekhinezh) [reply radix-engine/src/engine/call_frame.rs:1408]: Actually current `CallMethodWithAllResources` doesn't contain an argument

[the remark this answers is not in the document]
Comment (iamyulong) [reply radix-engine/src/engine/call_frame.rs:1384]: Oh right, haven't implemented the `ALL_WORKTOP_RESOURCES` :)

**Pull request merged**